### ISSUE TITLE
Value assigned to a variable in @BeforeSuite method is not reflecting properly in @Test methods (Raising new one as the previous issue is closed))

### ISSUE THREAD

**Issue by santhanam370** [cbeust/testng#2160]
### TestNG Version
> Note: only the latest version is supported

### Expected behavior
[testng] ...
   [testng] ... TestNG 7.0.0 by C??dric Beust (cedric@beust.com)
   [testng] ...
   [testng] 
   [testng] Dummy value before reinitializing in @BeforeSuite annoted method : Initial value
   [testng] Dummy value after reinitializing in @BeforeSuite annoted method : Reinitialized value
   [testng] Dummy value inside test method : **Reinitialized value**
   [testng] PASSED: test1
   [testng] 
   [testng] ===============================================
   [testng]     Regression1
   [testng]     Tests run: 1, Failures: 0, Skips: 0
   [testng] ===============================================
   [testng] 
   [testng] PASSED: test1
   [testng] 
   [testng] ===============================================
   [testng]     Regression2
   [testng]     Tests run: 1, Failures: 0, Skips: 0
   [testng] ===============================================
   [testng] 
   [testng] PASSED: test1
   [testng] 
   [testng] ===============================================
   [testng]     Regression3
   [testng]     Tests run: 1, Failures: 0, Skips: 0
   [testng] ===============================================
   [testng] 
   [testng] Dummy value inside test method : **Reinitialized value**
   [testng] PASSED: test1
   [testng] PASSED: test1
   [testng] 
   [testng] ===============================================
   [testng]     Regression2
   [testng]     Tests run: 2, Failures: 0, Skips: 0
   [testng] ===============================================
   [testng] 
   [testng] PASSED: test1
   [testng] PASSED: test1
   [testng] 
   [testng] ===============================================
   [testng]     Regression3
   [testng]     Tests run: 2, Failures: 0, Skips: 0
   [testng] ===============================================
   [testng] 
   [testng] Dummy value inside test method : Reinitialized value
   [testng] PASSED: test1
   [testng] PASSED: test1
   [testng] PASSED: test1
   [testng] 
   [testng] ===============================================
   [testng]     Regression3
   [testng]     Tests run: 3, Failures: 0, Skips: 0
   [testng] ===============================================
   [testng] 
   [testng] 
   [testng] ===============================================
   [testng] SampleTestSuite
   [testng] Total tests run: 3, Passes: 3, Failures: 0, Skips: 0
   [testng] ===============================================
   [testng] 

### Actual behavior

[testng] ... TestNG 7.0.0 by C??dric Beust (cedric@beust.com)
   [testng] ...
   [testng] 
   [testng] Dummy value before reinitializing in @BeforeSuite annoted method : Initial value
   [testng] Dummy value after reinitializing in @BeforeSuite annoted method : Reinitialized value
   [testng] Dummy value inside test method : Initial value
   [testng] PASSED: test1
   [testng] 
   [testng] ===============================================
   [testng]     Regression1
   [testng]     Tests run: 1, Failures: 0, Skips: 0
   [testng] ===============================================
   [testng] 
   [testng] PASSED: test1
   [testng] 
   [testng] ===============================================
   [testng]     Regression2
   [testng]     Tests run: 1, Failures: 0, Skips: 0
   [testng] ===============================================
   [testng] 
   [testng] PASSED: test1
   [testng] 
   [testng] ===============================================
   [testng]     Regression3
   [testng]     Tests run: 1, Failures: 0, Skips: 0
   [testng] ===============================================
   [testng] 
   [testng] Dummy value inside test method : Initial value
   [testng] PASSED: test1
   [testng] PASSED: test1
   [testng] 
   [testng] ===============================================
   [testng]     Regression2
   [testng]     Tests run: 2, Failures: 0, Skips: 0
   [testng] ===============================================
   [testng] 
   [testng] PASSED: test1
   [testng] PASSED: test1
   [testng] 
   [testng] ===============================================
   [testng]     Regression3
   [testng]     Tests run: 2, Failures: 0, Skips: 0
   [testng] ===============================================
   [testng] 
   [testng] Dummy value inside test method : Reinitialized value
   [testng] PASSED: test1
   [testng] PASSED: test1
   [testng] PASSED: test1
   [testng] 
   [testng] ===============================================
   [testng]     Regression3
   [testng]     Tests run: 3, Failures: 0, Skips: 0
   [testng] ===============================================
   [testng] 
   [testng] 
   [testng] ===============================================
   [testng] SampleTestSuite
   [testng] Total tests run: 3, Passes: 3, Failures: 0, Skips: 0
   [testng] ===============================================
   [testng]
### Is the issue reproductible on runner?

- [ ] Shell
- [ ] Maven
- [ ] Gradle
- [ Yes] Ant
- [ ] Eclipse
- [ ] IntelliJ
- [ ] NetBeans

### Test case sample
**TestNG xml config**
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
<suite name="SampleTestSuite" verbose="2" parallel="false" thread-count="1">
    <test name="Regression1">
        <packages>
            <package name="com.sample.*"/>
        </packages>
    </test>
    <test name="Regression2">
        <packages>
            <package name="com.sample.*"/>
        </packages>
    </test>
    <test name="Regression3">
        <packages>
            <package name="com.sample.*"/>
        </packages>
    </test>
</suite>

**Class containing test method :**
package com.sample;

import org.testng.annotations.BeforeSuite;
import org.testng.annotations.Test;

public class SampleTestNG {

    private String dummyVal = "Initial value";

    @BeforeSuite
    public void setup(){
        System.out.println("Dummy value before reinitializing in @BeforeSuite annoted method : " + dummyVal);
        dummyVal = "Reinitialized value";
        System.out.println("Dummy value after re
[TestNGBugInBeforeSuite.zip](https://github.com/cbeust/testng/files/3674646/TestNGBugInBeforeSuite.zip)
initializing in @BeforeSuite annoted method : " + dummyVal);
    }

    @Test
    public void test1(){
        System.out.println("Dummy value inside test method : " + dummyVal);
    }
}

**Comment by santhanam370** [comment]
Raised this one as the previous issue was closed https://github.com/cbeust/testng/issues/2158
Comments from #2158 
@krmahadevan :
This is not an issue with TestNG.

TestNG is working as designed.

@BeforeSuite is designed to be invoked only once per <suite>. Your suite xml contains three <test> tags. The setup() will get invoked for <test name="Regression1"> and gets skipped for <test name="Regression2"> and <test name="Regression3"> because they are all residing in the same <suite> tag.

If you need the initialization to happen for every <test> tag, please use @BeforeTest


@santhanam370 :
The problem here is, if we re-assign a value to a global variable inside @BeforeSuite method, it's not reflecting in @test method, only at the last test case execution the value is reflected.

You can see in the above logs, I have created a variable dummyVal with value "Initial value", and then reassigning the value inside @BeforeSuite method with value "Reinitialized value". Now our expectations is, inside the @test methods, the value of dummyVal should be "Reinitialized value". But out of execution of the test case for three times, in the first two times the value of dummyVal is "Initial value", only last execution of test case the "Reinitialized value" value is reflected.

I am able to replicate it for n times execution of test cases, the value reassigen inside @BeforeSuite is not reflecting for first (n-1) time, only in the nth time execution the correct value is reflecting

**Comment by krmahadevan** [comment]
Closing this as duplicate of https://github.com/cbeust/testng/issues/2158
